src/modules/role/role.service.ts

```diff
@@ -162,7 +162,7 @@ export class RoleService {
 
     if (!enrolmentPreconditions || enrolmentPreconditions.length < 1) return;
     for (const { type, conditions } of enrolmentPreconditions) {
-      if (type === 'role' && conditions) {
+      if (type === 'role' && conditions && conditions?.length > 0) {
```

Comment:
this is trivial but IMO, I think checking if the `conditions` variable exists and then doing `conditions?` defeats the purpose of optional chaining. 
we could just do `type === 'role' && conditions && conditions.length > 0` or
`type === 'role' &&  conditions?.length > 0`